File: Raven.Client.Lightweight/Linq/RavenQueryStatistics.cs
//-----------------------------------------------------------------------
// <copyright file="RavenQueryStatistics.cs" company="Hibernating Rhinos LTD">
//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------
using System;
using System.ComponentModel;
using Raven.Abstractions.Data;

namespace Raven.Client.Linq
{
	/// <summary>
	/// Statistics about a raven query.
	/// Such as how many records match the query
	/// </summary>
	public class RavenQueryStatistics
	{
		/// <summary>
		/// Whatever the query returned potentially stale results
		/// </summary>
		public bool IsStale { get; set; }

		/// <summary>
		/// What was the total count of the results that matched the query
		/// </summary>
		public int TotalResults { get; set; }

		/// <summary>
		/// Gets or sets the skipped results (duplicate documents);
		/// </summary>
		public int SkippedResults { get; set; }

		/// <summary>
		/// The time when the query results were unstale.
		/// </summary>
		public DateTime Timestamp { get; set; }

		/// <summary>
		/// The name of the index queried
		/// </summary>
		public string IndexName { get; set; }

		/// <summary>
		/// The timestamp of the queried index
		/// </summary>
		public DateTime IndexTimestamp { get; set; }

		/// <summary>
		/// The etag of the queried index
		/// </summary>
		public Guid IndexEtag { get; set; }

		/// <summary>
		/// Gets or sets a value indicating whether any of the documents returned by this query
		/// are non authoritative (modified by uncommitted transaction).
		/// </summary>
		public bool NonAuthoritativeInformation { get; set; }

		/// <summary>
		/// The timestamp of the last time the index was queried
		/// </summary>
		public bool LastQueryTime { get; set; }

		/// <summary>
		/// Update the query stats from the query results
		/// </summary>
		internal void UpdateQueryStats(QueryResult qr)
		{
			IsStale = qr.IsStale;
			NonAuthoritativeInformation= qr.NonAuthoritativeInformation;
			TotalResults = qr.TotalResults;
			SkippedResults = qr.SkippedResults;
			Timestamp = qr.IndexTimestamp;
			IndexName = qr.IndexName;
			IndexTimestamp = qr.IndexTimestamp;
			IndexEtag = qr.IndexEtag;
			//LastQueryTime = qr.LastQueryTime;
		}

		
	}
}

File: Raven.Client.Java/src/main/java/net/ravendb/client/RavenQueryStatistics.java
package net.ravendb.client;

import java.util.Date;

import net.ravendb.abstractions.data.Etag;
import net.ravendb.abstractions.data.QueryResult;


/**
 * Statistics about a raven query.
 * Such as how many records match the query
 */
public class RavenQueryStatistics {

  private boolean stale;
  private long durationMiliseconds;
  private int totalResults;
  private int skippedResults;
  private Date timestamp;
  private String indexName;
  private Date indexTimestamp;
  private Etag indexEtag;
  private boolean nonAuthoritativeInformation;
  private Date lastQueryTime;

  /**
   * Whatever the query returned potentially stale results
   * @return
   */
  public boolean isStale() {
    return stale;
  }

  /**
   * Whatever the query returned potentially stale results
   * @param stale
   */
  public void setStale(boolean stale) {
    this.stale = stale;
  }

  /**
   * The duration of the query _server side_
   * @return
   */
  public long getDurationMiliseconds() {
    return durationMiliseconds;
  }

  /**
   * The duration of the query _server side_
   * @param durationMiliseconds
   */
  public void setDurationMiliseconds(long durationMiliseconds) {
    this.durationMiliseconds = durationMiliseconds;
  }

  /**
   * What was the total count of the results that matched the query
   * @return
   */
  public int getTotalResults() {
    return totalResults;
  }

  /**
   * What was the total count of the results that matched the query
   * @param totalResults
   */
  public void setTotalResults(int totalResults) {
    this.totalResults = totalResults;
  }

  /**
   * Gets the skipped results
   * @return
   */
  public int getSkippedResults() {
    return skippedResults;
  }

  /**
   * Sets the skipped results
   * @param skippedResults
   */
  public void setSkippedResults(int skippedResults) {
    this.skippedResults = skippedResults;
  }


  /**
   * The time when the query results were unstale.
   * @return
   */
  public Date getTimestamp() {
    return timestamp;
  }

  /**
   * The time when the query results were unstale.
   * @param timestamp
   */
  public void setTimestamp(Date timestamp) {
    this.timestamp = timestamp;
  }

  /**
   * The name of the index queried
   * @return
   */
  public String getIndexName() {
    return indexName;
  }

  /**
   * The name of the index queried
   * @param indexName
   */
  public void setIndexName(String indexName) {
    this.indexName = indexName;
  }

  /**
   *  The timestamp of the queried index
   * @return
   */
  public Date getIndexTimestamp() {
    return indexTimestamp;
  }

  /**
   *  The timestamp of the queried index
   * @param indexTimestamp
   */
  public void setIndexTimestamp(Date indexTimestamp) {
    this.indexTimestamp = indexTimestamp;
  }

  /**
   * The etag of the queried index
   * @return
   */
  public Etag getIndexEtag() {
    return indexEtag;
  }

  /**
   * The etag of the queried index
   * @param indexEtag
   */
  public void setIndexEtag(Etag indexEtag) {
    this.indexEtag = indexEtag;
  }

  /**
   * Gets a value indicating whether any of the documents returned by this query
   * are non authoritative (modified by uncommitted transaction).
   * @return
   */
  public boolean isNonAuthoritativeInformation() {
    return nonAuthoritativeInformation;
  }

  /**
   * Sets a value indicating whether any of the documents returned by this query
   * are non authoritative (modified by uncommitted transaction).
   * @return
   */
  public void setNonAuthoritativeInformation(boolean nonAuthoritativeInformation) {
    this.nonAuthoritativeInformation = nonAuthoritativeInformation;
  }

  /**
   * The timestamp of the last time the index was queried
   * @return
   */
  public Date getLastQueryTime() {
    return lastQueryTime;
  }

  /**
   * The timestamp of the last time the index was queried
   * @param lastQueryTime
   */
  public void setLastQueryTime(Date lastQueryTime) {
    this.lastQueryTime = lastQueryTime;
  }



  public void updateQueryStats(QueryResult qr) {
    stale = qr.isStale();
    durationMiliseconds = qr.getDurationMiliseconds();
    nonAuthoritativeInformation = qr.isNonAuthoritativeInformation();
    totalResults = qr.getTotalResults();
    skippedResults = qr.getSkippedResults();
    timestamp = qr.getIndexTimestamp();
    indexName = qr.getIndexName();
    indexTimestamp = qr.getIndexTimestamp();
    indexEtag = qr.getIndexEtag();
  }

}
